esmvalcore/_data_finder.py

```diff
@@ -230,6 +230,21 @@ def _find_input_files(variable, rootpath, drs):
     return files
 
 
+def _filter_newest_files(input_files):
+    """Filter input files to get newest files if duplicates."""
+    if not len(input_files) > 1:
+        return input_files
+    else:
+        unique_newest_files = []
+        base_files = [os.path.basename(file) for file in input_files]
+        unique_base_files = list(set(base_files))
+        for unique_file in unique_base_files:
+            matches = fnmatch.filter(input_files, '*' + unique_file)
+            newest_match = max(matches, key=os.path.getctime)
```

Comment:
```python
            if matches:
                newest_match = sorted(matches)[-1]
                unique_newest_files.append(newest_match)
```
I would do something like this, I don't think the creation time is reliable, but the drs usually contains the version number somewhere. Make sure the ignore the 'latest' directory which is symlinked to the actual latest version on e.g. Jasmin.